runtime/lua/vim/lsp/callbacks.lua

```diff
@@ -37,6 +37,19 @@ end
 
 M['textDocument/references'] = function(_, _, result)
   if not result then return end
+  for _, loc in ipairs(result) do
```

Comment:
Wasn't sure if this could even be integrated in https://github.com/neovim/neovim/blob/ab12a229410c74e3bcc469bd34a3a4c40af8857c/runtime/lua/vim/lsp/util.lua#L736 there it would also affect other operations.

Reply:
I think we should move this operation to `M.locations_to_items`.
`M.locations_to_items` is used by only `M.set_loclist` and `M.set_qflist`.
I think this operation is useful for both.

Reply:
I've pushed another commit that integrates it into `M.locations_to_items` - made the change also a lot simpler because there `line` is already accessible.